File: app/src/main/java/vinhlong/ditagis/com/qlcln/entities/DApplication.java
package vinhlong.ditagis.com.qlcln.entities;

import android.app.Application;
import android.location.Location;

import com.esri.arcgisruntime.data.ArcGISFeature;
import com.esri.arcgisruntime.geometry.Geometry;


import java.net.URISyntaxException;

import io.socket.client.IO;
import io.socket.client.Socket;
import vinhlong.ditagis.com.qlcln.entities.entitiesDB.User;
import vinhlong.ditagis.com.qlcln.utities.Constant;

public class DApplication extends Application {



    private User user;

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }


    private short loaiVatTu;

    public short getLoaiVatTu() {
        return loaiVatTu;
    }

    public void setLoaiVatTu(short loaiVatTu) {
        this.loaiVatTu = loaiVatTu;
    }


    private Geometry geometry;

    public Geometry getGeometry() {
        return geometry;
    }

    public void setGeometry(Geometry geometry) {
        this.geometry = geometry;
    }

    private ArcGISFeature arcGISFeature;

    public ArcGISFeature getArcGISFeature() {
        return arcGISFeature;
    }

    public void setArcGISFeature(ArcGISFeature arcGISFeature) {
        this.arcGISFeature = arcGISFeature;
    }


    private Location mLocation;

    public Location getmLocation() {
        return mLocation;
    }

    public void setmLocation(Location mLocation) {
        this.mLocation = mLocation;
    }
}

File: app/src/main/java/vinhlong/ditagis/com/qlcln/entities/DApplication.kt
package vinhlong.ditagis.com.qlcln.entities


import android.app.Application
import android.graphics.Bitmap
import android.location.Location
import com.esri.arcgisruntime.data.Attachment
import com.esri.arcgisruntime.data.Feature
import com.esri.arcgisruntime.data.ServiceFeatureTable
import com.esri.arcgisruntime.geometry.Point
import com.esri.arcgisruntime.layers.FeatureLayer
import vinhlong.ditagis.com.qlcln.libs.FeatureLayerDTG
import vinhlong.ditagis.com.qlcln.utities.Constant
import vinhlong.ditagis.com.qlcln.utities.DProgressDialog

class DApplication : Application() {
    var progressDialog = DProgressDialog()
    var statusCode = Constant.StatusCode.NORMAL.value
    var appInfo: DAppInfo? = null
    var layerInfos: List<DLayerInfo>? = null
    var center: Point? = null
    var bitmaps: ArrayList<Bitmap>? = null
    var selectedAttachment: Attachment? = null
    var selectedBitmap: Bitmap? = null
    var user: User? = null
    var selectedFeature: Feature? = null
    private var mLocation: Location? = null
    var diemDanhGia: FeatureLayerDTG? = null
    var mauKiemNghiem: FeatureLayerDTG? = null
    var serviceFeatureTableHanhChinh: ServiceFeatureTable? = null
    fun setmLocation(mLocation: Location) {
        this.mLocation = mLocation
    }
}